src/Servers/Kestrel/Transport.Quic/test/WebHostTests.cs

```diff
@@ -152,6 +152,73 @@ public async Task Listen_Http3AndSocketsCoexistOnSameEndpoint_ClientSuccess()
             await host.StopAsync().DefaultTimeout();
         }
 
+        [ConditionalFact]
+        [MsQuicSupported]
+        public async Task Listen_Http3AndSocketsCoexistOnSameEndpoint_AltSvcEnabled_Upgrade()
+        {
+            // Arrange
+            var builder = GetHostBuilder()
```

Comment:
Doesn't the ALPN setting need to be updated to match?
https://github.com/dotnet/aspnetcore/blob/836d664bee463d7ceb3312df39296151f8b06d0d/src/Servers/Kestrel/Transport.Quic/test/QuicTestHelpers.cs#L28

Reply:
HttpClient is being inconsistent and using h3 in the altsvc but not as it’s alpn https://github.com/dotnet/runtime/issues/55894?notification_referrer_id=MDE4Ok5vdGlmaWNhdGlvblRocmVhZDIxODY4Nzc0MTY6MzAzMjAx#issuecomment-882839687

Reply:
Ok. The alt-svc and ALPN should be kept in sync, let's cover this later today.